app/services/deadline_calculator/calendar_days.rb

```diff
@@ -24,12 +24,24 @@ def external_deadline
       kase.received_date + kase.correspondence_type.external_time_limit.days
     end
 
+    def extension_deadline(time_limit)
+      kase.received_date + (time_limit + kase.correspondence_type.external_time_limit).days
+    end
+    
+    def max_allowed_deadline_date(time_limit=nil)
+      time_limit ||= (kase.correspondence_type.extension_time_limit || 0)
```

Comment:
There's something weird about this method. At first I was looking at it thinking, can we get rid of the second, nested conditional here? But actually, I'm a bit weirded out by the fact this seems to require time_limit and yet it's an optional argument.

So we have three cases?

1. Pass in a time_limit - it just gets used
2. if time_limit omitted from the method call, default to the extension time_limit from the kase's correspondence type
3. if there's no extension_time_limit from there, use `0`

It feels like this method knows too much about other things - would it be a daft idea to think about correspondence_type.extension_time_limit always returning 0 if it's unset?

Reply:
yes what you guess is correct, if correspondence_type.extension_time_limit is unset,  I think it will return nil?